optuna/structs.py

```diff
@@ -60,12 +60,13 @@ class FrozenTrial(
             ('system_attrs', Dict[str, Any]),
             ('intermediate_values', Dict[int, float]),
             ('params_in_internal_repr', Dict[str, float]),
+            ('trial_id', int),
         ])):
     """Status and results of a :class:`~optuna.trial.Trial`.
 
     Attributes:
-        trial_id:
-            Identifier of the :class:`~optuna.trial.Trial`.
+        number:
```

Comment:
Just in my opinion, it would be worth clarifying in the document that `number` is zero-based indexing and `trial_id` depends on storage backend.

Reply:
Sounds good. I added the explanation. Thank you.